lib/suspenders/actions.rb

```diff
@@ -81,5 +87,178 @@ def hash_unmerge(hash, subhash)
         end
       end
     end
+
+    class Gem
+      def initialize(destination_root, name, version, **options)
+        @destination_root = destination_root
+        @name = name
+        @version = version
+        @groups = Array(options.delete(:group))
+        @options = options
+      end
+
+      def invoke!
+        existing_gemfile = IO.read(gemfile_path)
```

Comment:
I was trying to avoid doing IO in the initializer. But maybe it's fine so long as I'm not, like, writing to disc in the initializer.

Reply:
Yeah it's a pretty good reason and it's already this way so your choice on the matter. It would certainly help the arity of the private methods stay to <=2 but at the end of the day it's minor.

Reply:
These are all good reasons. One option is to pass the IO object to the initializer, but not very practical. Maybe we could do something like this:

```ruby
def existing_gemfile
  @existing_gemfile ||= IO.read(gemfile_path)
end
```

Which incurs the IO cost only once and avoids duplication (at the expense of "where does this came from?").